lib/install/management/appliance.go

```diff
@@ -505,20 +558,21 @@ func (d *Dispatcher) createAppliance(conf *config.VirtualContainerHostConfigSpec
 
 	// Create the VCH inventory folder
 	vchFolder := d.session.VMFolder
+	folderPath := fmt.Sprintf("%s/%s", d.session.VMFolder.InventoryPath, conf.Name)
 	if d.isVC {
 		d.op.Info("Creating the VCH folder")
 
 		vchFolder, err = d.session.VMFolder.CreateFolder(d.op, spec.Name)
 		if err != nil {
-			d.op.Debugf("Encountered unexpected error: %s", err)
 			if f, ok := err.(types.HasFault); ok {
-				if _, ok = f.Fault().(*types.DuplicateName); ok {
-					return fmt.Errorf("an object with the same name as the VCH folder (%s) already exists in that path", spec.Name)
+				switch f.Fault().(type) {
```

Comment:
why?

Reply:
Moved to the switch case because it cut's down on code additions. I felt it made the code more readible when I had the composite case declaration along with the default case.

Reply:
`CreateFolder` will only throw `DuplicateName` -- `AlreadyExists` is not an error we will encounter in this path, so I don't think this is needed.

Reply:
I was running into this error when reenabling the test, that is why I have added it here. I saw that it can be thrown from the vsphere api's, but I was not sure how that translated to govmomi.